csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/AssertUtils.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.ML.OnnxRuntime.Tests
+{
+  internal static class AssertUtils
+  {
+
+    public static void IfThrowsCheckException<T>(Action action, string feedbackMessage, string expectedExceptionMessage = null) where T : Exception
+    {
+      try
+      {
+        action();
+      }
+      catch (T ex)
+      {
+        if (expectedExceptionMessage == null)
+        {
+          return;
+        }
+        else
+        {
+          Assert.Contains(expectedExceptionMessage, ex.Message);
+        }
+      }
+      catch (Exception ex)
+      {
+        Assert.Fail($"{feedbackMessage}\nExpected {typeof(T).Name} but got {ex.GetType().Name}. S");
```

Comment:
there's a trailing "S".

also, it seems inconsistent to only output `feedbackMessage` in this assertion and not the expectedExceptionMessage one. does an assertion failure show the callstack? if so, maybe we can just not have `feedbackMessage`.

Reply:
Ah the assertion failure only gives the feedback that an exception of type T was expected but none was thrown, so I wrote the assertion methods to provide some kind of feedback. 

It took a little bit of time to narrow down which line was failing in the unit test because the unit test checks for multiple assertions.

I didn't add any kind of feedback message to the expected exception message check because the expected exceptions are usually more detailed, but I would be open to adding the feedback message to the check for the exception message as well.